### PR TITLE
Added Colemak as a persistent layer for the minidox

### PR DESCRIPTION
**Pull request by djuggler**
I modified the default minidox keyboard layout to:

- Have the backspace on the left thumb key (KC_BSPC)
- Have the space on the right thumb key (KC_SPC)
- Use lower+right_thumb as enter (KC_ENT)
- Added arrow keys to the right half of the keyboard with the adjust layer (KC_LEFT, KC_DOWN, KC_RGHT on lower right row and KC_UP on home row)

### CODE REVIEW

**Comment by drashna** [comment]
You need to run `make git-submodule` and add a new commit to fix the compiling issue (changes to ChibiOS).

File: keyboards/minidox/keymaps/djuggler/config.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/minidox/keymaps/djuggler/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/minidox/keymaps/djuggler/config.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#define CONFIG_USER_H
+
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/minidox/keymaps/djuggler/config.h]: This line is not actually needed, and should be removed. 

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/minidox/keymaps/djuggler/keymap.c [hunk 3]
```diff
@@ -0,0 +1,200 @@
+#include "minidox.h"
```
Comment (drashna) [inline keyboards/minidox/keymaps/djuggler/keymap.c]: Could you replace this include with `#include QMK_KEYBOARD_H` instead?

File: keyboards/minidox/keymaps/djuggler/keymap.c [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/minidox/keymaps/djuggler/keymap.c]: Could you remove all of these additional includes?  

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/minidox/keymaps/djuggler/keymap.c [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  COLEMAK,
+  LOWER,
+  RAISE,
+  ADJUST,
+};
+
+// Fillers to make layering more clear
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/minidox/keymaps/djuggler/keymap.c]: Could you remove these two defines?  

They're included in QMK's core files, so are redundant here.

File: keyboards/minidox/keymaps/djuggler/keymap.c [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+ */
+[_ADJUST] = LAYOUT(\
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,        KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10, \
+  KC_F11,  KC_F12,  QWERTY, RGB_SAI, RGB_SAD,      RGB_VAD, COLEMAK,   KC_UP, TSKMGR, CALTDEL, \
+  RESET,   RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD,      RGB_VAI, KC_LEFT, KC_DOWN, KC_RGHT,  _______, \
+                    _______, _______, _______,      _______,  _______, _______                    \
+)
+};
+
+void persistant_default_layer_set(uint16_t default_layer) {
```
Comment (drashna) [inline keyboards/minidox/keymaps/djuggler/keymap.c]: Could you remove this function?  

It's been depreciated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/minidox/keymaps/djuggler/keymap.c [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
+          PLAY_SONG(tone_qwerty);
+        #endif
+        persistant_default_layer_set(1UL<<_QWERTY);
```
Comment (drashna) [inline keyboards/minidox/keymaps/djuggler/keymap.c]: This should be replaced with `set_single_persistent_default_layer(_QWERTY)`

File: keyboards/minidox/keymaps/djuggler/keymap.c [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+      }
+      return false;
+      break;
+      case COLEMAK:
+        if (record->event.pressed) {
+          #ifdef AUDIO_ENABLE
+            PLAY_SONG(tone_colemak);
+          #endif
+          persistant_default_layer_set(1UL<<_COLEMAK);
```
Comment (drashna) [inline keyboards/minidox/keymaps/djuggler/keymap.c]: This should be replaced with `set_single_persistent_default_layer(_COLEMAK)`

File: keyboards/minidox/keymaps/djuggler/rules.mk [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# Bootloader
+#     This definition is optional, and if your keyboard supports multiple bootloaders of
+#     different sizes, comment this out, and the correct address will be loaded
+#     automatically (+60). See bootloader.mk for all options.
+BOOTLOADER = caterina
+
+RGBLIGHT_ENABLE = yes        # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
+
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/minidox/keymaps/djuggler/rules.mk]: Could you remove this (ifndef/include/endif) block? 

It's not needed, and is a holdover from the old build system.

File: keyboards/planck/keymaps/djuggler/config.h [hunk 10]
```diff
@@ -0,0 +1,42 @@
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/planck/keymaps/djuggler/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/planck/keymaps/djuggler/config.h [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+#define CONFIG_USER_H
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/planck/keymaps/djuggler/config.h]: This line is not actually needed, and should be removed. 

Specifically, this should be included in the keyboard's `config.h` file, and is redundant.

File: keyboards/planck/keymaps/djuggler/keymap.c [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "planck.h"
```
Comment (drashna) [inline keyboards/planck/keymaps/djuggler/keymap.c]: Could you replace this include with `#include QMK_KEYBOARD_H` instead?

File: keyboards/planck/keymaps/djuggler/keymap.c [hunk 13]
```diff
@@ -0,0 +1,264 @@
...
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/planck/keymaps/djuggler/keymap.c]: Could you remove all of these additional includes?  

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/planck/keymaps/djuggler/keymap.c [hunk 14]
```diff
@@ -0,0 +1,264 @@
...
+ * | Tab  |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Bksp |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Esc  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  |Enter |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Brite| Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = {
```
Comment (drashna) [inline keyboards/planck/keymaps/djuggler/keymap.c]: This should be updated to the `LAYOUT_planck_grid` or `LAYOUT_ortho_4x12`.

https://github.com/qmk/qmk_firmware/blob/master/keyboards/planck/keymaps/default/keymap.c#L57-L62

Comment (djuggler) [reply keyboards/planck/keymaps/djuggler/keymap.c]: Thank you. Will get it done.

**Comment by drashna** [comment]
Any update on this?

**Comment by djuggler** [comment]
Yes. I'm working on implementing your requests. My apologies on the delay.
I have been covered up with work and family.

This weekend is already scheduled but I can probably have the changes taken
care of by end of next weekend (say by Feb 4).

Thanks.

Doug McCaughan
"I solve problems"
+1-865-898-7189
Skype:djuggler

**Comment by drashna** [comment]
Awesome, and no worries.  I just wanted to touch base here, since I was going through all the PRs.

**Comment by drashna** [comment]
Have you had a chance to update this?

**Comment by djuggler** [comment]
No, I was part of the Knoxville, TN floods on Feb 23 and have spent each
weekend cleaning up the basement. I will make a note to myself to wrap up
the qmk changes this weekend without fail.

Thank you for your patience.
Doug
"I solve problems"
+1-865-898-7189
Skype:djuggler

**Comment by drashna** [comment]
I'm very sorry to hear that!  

And no worries!  Feel free to take your time.

**Comment by drashna** [comment]
Just touching bases here, have you had a chance to work on this?

**Comment by djuggler** [comment]
I actually have availability in my schedule and commit to you to have this
completed by Sunday night (Aug 4).

Thank you for following up.

Doug McCaughan
"I solve problems"
+1-865-898-7189
Skype:djuggler

File: keyboards/minidox/keymaps/djuggler/keymap.c [hunk 15]
```diff
@@ -0,0 +1,200 @@
...
+#include "action_layer.h"
+#include "eeconfig.h"
+
+extern keymap_config_t keymap_config;
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+#define _QWERTY 0
```
Comment (noroadsleft) [inline keyboards/minidox/keymaps/djuggler/keymap.c]: Lines 11-15 should be replaced with:

```c
enum layer_names {
	_QWERTY,
	_COLEMAK,
	_LOWER,
	_RAISE,
	_ADJUST,
};
```

Having the _RAISE layer at 3 and _ADJUST at 16 will make the compiler insert empty layers at each index from 4 to 15, increasing firmware size for no change in functionality.

**Comment by drashna** [comment]
Have you had a chance to update this PR?

**Comment by djuggler** [comment]
I'll do it tomorrow without fail.

Doug
"I solve problems"
+1-865-898-7189
Skype:djuggler

**Comment by drashna** [comment]
just wanted to touch base here. 

and i think we may close this PR for now.  And if we do, you can reopen the PR or open a new PR when you're ready.

**Comment by djuggler** [comment]
If you give me the weekend, I'll have it done by Sunday evening. If I
don't, go ahead and close it. Sorry I've kept putting this off. Been crazy
busy but this weekend is open.

Doug
"I solve problems"
+1-865-898-7189
Skype:djuggler

**Comment by drashna** [comment]
Sounds good! 

And no worries, life happens! But we've been wanting to clean up stuff, so it's easier for us to go through PRs, so we can get to them quicker.  

If you need any help, don't hesitate to ask.

**Pull request closed without merging**